apps/jobs/views.py

```diff
@@ -655,129 +531,30 @@ def get_all_entries_on_public_leaderboard(request, challenge_phase_split_pk):
-    # Exclude the submissions from challenge host team to be displayed on the leaderboard of public phases
-    challenge_hosts_emails = (
-        [] if not is_challenge_phase_public else challenge_hosts_emails
+    distinct_sorted_leaderboard_data, http_status_code = calculate_distinct_sorted_leaderboard_data(
+        request.user,
+        challenge_obj,
+        challenge_phase_split,
+        only_public_entries=False,
     )
-
-    all_banned_email_ids = challenge_obj.banned_email_ids
-
-    leaderboard_data = LeaderboardData.objects.exclude(
-        Q(submission__created_by__email__in=challenge_hosts_emails)
-        & Q(submission__is_baseline=False)
-    )
-    # Get all the successful submissions related to the challenge phase split
-    leaderboard_data = leaderboard_data.filter(
-        challenge_phase_split=challenge_phase_split,
-        submission__status=Submission.FINISHED,
-    ).order_by("-created_at")
-
-    leaderboard_data = leaderboard_data.annotate(
-        filtering_score=RawSQL(
-            "result->>%s", (default_order_by,), output_field=FloatField()
-        ),
-        filtering_error=RawSQL(
-            "error->>%s",
-            ("error_{0}".format(default_order_by),),
-            output_field=FloatField(),
-        ),
-    ).values(
-        "id",
-        "submission__participant_team",
-        "submission__participant_team__team_name",
-        "submission__participant_team__team_url",
-        "submission__is_baseline",
-        "challenge_phase_split",
-        "result",
-        "error",
-        "filtering_score",
-        "filtering_error",
-        "leaderboard__schema",
-        "submission__submitted_at",
-        "submission__method_name",
-        "submission__is_public"
-    )
-    all_banned_participant_team = []
-    for leaderboard_item in leaderboard_data:
-        participant_team_id = leaderboard_item["submission__participant_team"]
-        participant_team = ParticipantTeam.objects.get(id=participant_team_id)
-        all_participants_email_ids = (
-            participant_team.get_all_participants_email()
-        )
-        for participant_email in all_participants_email_ids:
-            if participant_email in all_banned_email_ids:
-                all_banned_participant_team.append(participant_team_id)
-                break
-        if leaderboard_item["error"] is None:
-            leaderboard_item.update(filtering_error=0)
-
-    if challenge_phase_split.show_leaderboard_by_latest_submission:
-        sorted_leaderboard_data = leaderboard_data
-    else:
-        sorted_leaderboard_data = sorted(
-            leaderboard_data,
-            key=lambda k: (
-                float(k["filtering_score"]),
-                float(-k["filtering_error"]),
-            ),
-            reverse=True
-            if challenge_phase_split.is_leaderboard_order_descending
-            else False,
+    if http_status_code == status.HTTP_400_BAD_REQUEST:
```

Comment:
Same. Please add a comment before this line to explain action to be taken here.

Reply:
The action will be returned by the function.